cmake/ensure-executable-naming-convention.sh.in

```diff
@@ -0,0 +1,54 @@
+#
+# Ensure (part of) our executable naming convention
+# (see https://github.com/AliceO2Group/CodingGuidelines)
+#
+# what we test here is that all the names of the executables
+# in @CMAKE_INSTALL_PREFIX@/bin/ :
+# - are all lowercases
+# - contain no special character other than "-"
+# - start with o2- (or test)
+# - (unless it's a test) is of the form o2-xxx
+# where xxx is in a list of known sub-systems for the AliceO2 repository
+#
+
+rc=0
+
+error_msg() {
+        echo "ERROR: executable name $1 does not follow naming convention : $2"
+        rc=1
+}
+
+warning_unknown_subsystem() {
+        echo "WARNING: in $1 the subsystem $2 is not in the list of know subsystem"
+}
+
+re_is_test="^test"
+re_start_with_o2_or_test="^o2-"
+re_only_lower_case_or_dash="^([a-z,0-9]|-)+$"
```

Comment:
This actually allows lowercase letters, digits and the comma. Correct regex:

```
^[a-z0-9]([a-z0-9]+|-)+[a-z0-9]$
```

This one also ensures the executable is at least three characters long. (Unfortunately Bash regexp do not support non-greedy matching afaik.)